annotate.el

```diff
@@ -1073,6 +1090,116 @@ example:
+               (checksum        (annotate-checksum-from-dump file-matched-record))
+               (new-record      (annotate-make-record record-filename
+                                                      new-annotations
+                                                      checksum)))
+          (push new-record
+                rest-of-db))
+      db-records)))
+
+(defun annotate-db-replace-annotation (db-records
+                                       record-filename
+                                       annotation-beginning
+                                       annotation-ending
+                                       replacing-text)
+  "TODO"
+  (let* ((filename-match-p (lambda (record)
+                             (string= (annotate-filename-from-dump record)
+                                      record-filename)))
+         (file-matched-record (cl-find-if filename-match-p db-records)))
+    (if file-matched-record
+        (let ((old-annotation   (cl-find-if (lambda (a)
+                                              (and (= (annotate-beginning-of-annotation a)
+                                                      annotation-beginning)
+                                                   (= (annotate-ending-of-annotation    a)
+                                                      annotation-ending)))
+                                            (annotate-annotations-from-dump file-matched-record))))
+             (if old-annotation
+                 (let* ((rest-of-db       (cl-remove-if filename-match-p db-records))
+                        (rest-annotations (cl-remove-if (lambda (a)
+                                                          (and (= (annotate-beginning-of-annotation a)
+                                                                  annotation-beginning)
+                                                               (= (annotate-ending-of-annotation    a)
+                                                                  annotation-ending)))
+                                                        (annotate-annotations-from-dump file-matched-record)))
+                        (checksum        (annotate-checksum-from-dump file-matched-record))
+                        (new-annotation  (annotate-make-annotation annotation-beginning
+                                                                   annotation-ending
+                                                                   replacing-text
+                                                                   (annotate-annotated-text old-annotation)))
+                        (new-record      (annotate-make-record record-filename
+                                                               (append (list new-annotation)
+                                                                       rest-annotations)
+                                                               checksum)))
+                   (push new-record
```

Comment:
Why is `new-record` pushed on both the `rest-annotations` and `rest-of-db`? Or should the second `push` push `rest-annotations` onto `rest-of-db`? It seems to me that we are dropping `rest-annotations` here.

Reply:
Hi Bastian!
Ouch! That form:

```
  (push new-record
                       rest-annotations)
```

was just wrong. :/
Bye!
C.